network/network.go

```diff
@@ -514,9 +516,8 @@ func (n *network) PushQuery(validatorIDs ids.ShortSet, chainID ids.ID, requestID
 			err,
 			len(container))
 		n.log.Verbo("container: %s", formatting.DumpBytes{Bytes: container})
-		for _, validatorID := range validatorIDs.List() {
-			vID := validatorID
-			n.executor.Add(func() { n.router.QueryFailed(vID, chainID, requestID) })
+		for validatorIDKey := range validatorIDs {
+			n.executor.Add(func() { n.router.QueryFailed(ids.NewShortID(validatorIDKey), chainID, requestID) })
```

Comment:
This doesn't work as expected anymore, we can't pass a loop variable into an anonymous function. It'll overwrite the value. This should be more in line with the previous change:
```
validatorID := ids.NewShortID(validatorIDKey)
// use validatorID inside the lambda function
```